thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/dashboard/resources/EmailResource.java

```diff
@@ -122,11 +123,17 @@ public Response generateAndSendAlertForDatasets(@PathParam("startTime") Long sta
     if (Strings.isNullOrEmpty(toAddr)) {
       throw new WebApplicationException("Empty : list of recipients" + toAddr);
     }
-    if(Strings.isNullOrEmpty(teHost)) {
-      throw new WebApplicationException("Invalid TE host" + teHost);
+
+    SmtpConfiguration smtpConfiguration = thirdeyeConfiguration.getSmtpConfiguration();
+    if (!Strings.isNullOrEmpty(smtpHost)) {
+      smtpConfiguration.setSmtpHost(smtpHost);
     }
-    if (Strings.isNullOrEmpty(smtpHost)) {
-      throw new WebApplicationException("invalid smtp host" + smtpHost);
+    if (smtpPort != null) {
```

Comment:
At line 127, smtpConfig has been assigned to thirdeyeConfiguration.getSmtpConfiguration();